app/models/registration_system.rb

```diff
@@ -121,10 +121,10 @@ def self.database_options
   def self.write_rhsm_config(params)
     FileUtils.copy(RHSM_CONFIG_FILE, "#{RHSM_CONFIG_FILE}.miq_orig") unless File.exist?("#{RHSM_CONFIG_FILE}.miq_orig")
     rhsm_config = File.read(RHSM_CONFIG_FILE)
-    rhsm_config[/(\s*)proxy_hostname(\s*)=(.*)/, 3] = " #{params[:proxy_hostname]}"
-    rhsm_config[/(\s*)proxy_port(\s*)=(.*)/, 3]     = " #{params[:proxy_port]}"
-    rhsm_config[/(\s*)proxy_user(\s*)=(.*)/, 3]     = " #{params[:proxy_user]}"
-    rhsm_config[/(\s*)proxy_password(\s*)=(.*)/, 3] = " #{params[:proxy_password]}"
+    rhsm_config[/\s*?proxy_hostname\s*?=(.*?)$/, 1] = " #{params[:proxy_hostname]}"
```

Comment:
I am 👍 for the changes with the capture groups, but unsure I understand the need for the `non-greedy` operators at all.

For the `\s` ones, those should already be fine as is (as far as I know) since they are already scoped to "whitespace" chars, and I recall that Ruby's regexp engine works within the context of a single line by default, but I could be wrong.

I see that we are adding the `$` as part of this, so I think that should be sufficient so the `(.*)` bit doesn't need a non-greedy (`?`) operator.

Unless there is a performance benefit or other reason I am not aware of, I feel like the extra operators will just muddy the water for the less "`Regexp`-inclined" developers out there.

Reply:
A simple illustration of my point:

```console
$ cat config.conf
foo=bar
bar=bar
baz=bar
qux=bar
$ irb
irb(main):001:0> data = IO.read 'config.conf'
=> "foo=bar\nbar=bar\nbaz=bar\nqux=bar\n"
irb(main):002:0> puts data
foo=bar
bar=bar
baz=bar
qux=bar
=> nil
irb(main):003:0> data[/\s*bar\s*=(.*)/]
=> "\nbar=bar"
irb(main):004:0> data[/\s*bar\s*=(.*)/, 1]
=> "bar"
irb(main):005:0>
```

Reply:
To be more specific, I think you need the `Regexp::MULTILINE`:

https://github.com/ruby/ruby/blob/3679023df822d8bf3f88b9b2e4b849318d2c409e/doc/regexp.rdoc#options

For this to be a problem with new lines:

```console
irb(main):003:0> data[/\s*bar\s*=(.*)/]
=> "\nbar=bar"
irb(main):004:0> data[/\s*bar\s*=(.*)/, 1]
=> "bar"
irb(main):005:0> data[/\s*bar\s*=(.*)/m]
=> "\nbar=bar\nbaz=bar\nqux=bar\n"
irb(main):006:0> data[/\s*bar\s*=(.*)/m, 1]
=> "bar\nbaz=bar\nqux=bar\n"
irb(main):007:0>
```